### DISCUSSION TITLE
Support downloading a single file app to specified directory?

### DISCUSSION THREAD

**Discussion by Konecho** [ScoopInstaller/Scoop#5181] [Ideas]
Some applications have only a single executable and will read the configuration in the execution directory by default, these applications need to be placed in a specific directory to work properly. An example is [winsw](https://github.com/winsw/winsw), a wrapper executable that can run any executable as a Windows service. Also, many apps written in golang or rust (such as [alist](https://github.com/alist-org/alist)) are compiled into single executables. scoop buckets will never contain these apps, but these apps need a convenient way to be downloaded, imagine we can use `scoop get xxx` (or something else) to generate a latest verion copy of `xxx.exe` in the current directory.

**Comment by niheaven** [comment]
> scoop buckets will never contain these apps

? You could add these apps into main bucket, and `winsw` is there (https://github.com/ScoopInstaller/Main/blob/master/bucket/winsw.json)

**Reply by Konecho** [reply]
It does have. The reason I mention winsw as an example is that one of its usages is to rename the exe to an app name and place it in the same directory as the xml config file. While apps like winsw can be used as a global command line tool, for some other apps it doesn't make much sense to put the executable in the `apps` directory of scoop.

If this idea wasn't a common demand the only thing I'd probably need to do is write a powershell script to download the Install package directly to the current directory via scoop instead of installing it.

**Reply by rashil2000** [reply]
> for some other apps it doesn't make much sense to put the executable in the apps directory of scoop.

This is extremely rare. WinSW is the only tool that gets used this way (i.e. multiple copies in different directories). 

> If this idea wasn't a common demand the only thing I'd probably need to do is write a powershell script to download the Install package directly to the current directory via scoop instead of installing it.

Yeah, you can simply use a `curl` + `tar` combo to do it, i.e. `curl <url> | tar -<args>`

**Reply by Konecho** [reply]
> Yeah, you can simply use a `curl` + `tar` combo to do it, i.e. `curl <url> | tar -<args>`

Yes, and I realized that the scoop command already did most of the work I needed to do (downloading the latest installer to the cache directory, hash checking), I just needed to define a new command to move it from the cache directory to the current directory.

**Reply by niheaven** [reply]
You mean `scoop download`? That command only download the apps and hash check, and the apps are in cache folder.

**Reply by Konecho** [reply]
> You mean `scoop download`? That command only download the apps and hash check, and the apps are in cache folder.

Yep, that's almost enough.
